tests/integration/test_task.py

```diff
@@ -80,3 +82,219 @@ def _run(self, input_files):
+    script_file = os.path.abspath(os.path.join(diagnostics_root, diag_script))
+    ept = \
+        "Cannot execute script diag_cow.py ({}): file does not exist.".format(
+            script_file)
+    assert ept == str(err_msg.value)
+
+
+def _get_diagnostic_tasks(tmp_path, diagnostic_text, extension):
+    """Assemble Python diagnostic tasks of DiagnosticTasks."""
+    diag = 'diag_cow.' + extension
+    diag_script = tmp_path / diag
+    diag_output_dir = tmp_path / 'mydiag'
+    diag_run_dir = diag_output_dir / 'run_dir'
+    diag_plot_dir = diag_output_dir / 'plot_dir'
+    diag_work_dir = diag_output_dir / 'work_dir'
+    diag_settings = {'run_dir': diag_run_dir.as_posix(),
+                     'plot_dir': diag_plot_dir.as_posix(),
+                     'work_dir': diag_work_dir.as_posix(),
+                     'profile_diagnostic': False,
+                     'exit_on_ncl_warning': False}
+
+    with open(diag_script, "w") as fil:
+        fil.write(diagnostic_text)
+    tasks = set()
+    task = DiagnosticTask(
+        name='task0',
+        ancestors=None,
+        script=diag_script.as_posix(),
+        settings=diag_settings,
+        output_dir=diag_output_dir.as_posix(),
+    )
+    for task0 in task.flatten():
+        task0.priority = 0
+    tasks.add(task)
+
+    return tasks
+
+
+def test_py_diagnostic_run_sequential_task_fails(monkeypatch, tmp_path):
+    """Run DiagnosticTask sequentially with bad Python script."""
+    diagnostic_text = "import os\n\nprint(cow)"
+    tasks = _get_diagnostic_tasks(tmp_path, diagnostic_text, 'py')
+
+    def _run(self, input_filesi=[]):
+        print(f'running task {self.name}')
+
+    monkeypatch.setattr(BaseTask, '_run', _run)
+
+    with pytest.raises(DiagnosticError) as err_mssg:
+        _run_tasks_sequential(tasks)
+    exp_mssg = "diag_cow.py failed with return code 1"
+    assert exp_mssg in str(err_mssg.value)
+
+
+def test_py_diagnostic_run_parallel_task_fails(monkeypatch, tmp_path):
+    """Run DiagnosticTask parallel with bad Python script."""
+    diagnostic_text = "import os\n\nprint(cow)"
+    tasks = _get_diagnostic_tasks(tmp_path, diagnostic_text, 'py')
+
+    def _run(self, input_filesi=[]):
+        print(f'running task {self.name}')
+
+    monkeypatch.setattr(BaseTask, '_run', _run)
+
+    with pytest.raises(DiagnosticError) as err_mssg:
+        _run_tasks_parallel(tasks, 2)
+    exp_mssg = "diag_cow.py failed with return code 1"
+    assert exp_mssg in str(err_mssg.value)
+
+
+def test_py_diagnostic_run_parallel_task(monkeypatch, tmp_path):
+    """Run DiagnosticTask in parallel with OK Python script."""
+    diagnostic_text = "import os\n\nprint('cow')"
+    tasks = _get_diagnostic_tasks(tmp_path, diagnostic_text, 'py')
+
+    def _run(self, input_filesi=[]):
+        print(f'running task {self.name}')
+
+    monkeypatch.setattr(BaseTask, '_run', _run)
+
+    _run_tasks_parallel(tasks, 2)
+
+
+@pytest.mark.install
```

Comment:
You could probably just monkeypatch `which` so it returns a fake path to the NCL executable instead of using the install marker, since you're not actually going to run the NCL executable

Reply:
no that actually runs the ncl executable via the diagnostic task, hence the two outcome cases - fail and warning

Reply:
I see. But why are you monkey patching `BaseTask._run`? If you would patch `DiagnosticTask._run` instead, there would be no need to run the NCL interpreter.

Reply:
I actually want to see the running of the task ie running the executables (python, ncl, R) with a small basic diag as part of the test - is that beyond the scope of this test purpose you reckon?

Reply:
I think that's fine, but it would be really good to also have a version that doesn't require all these extra's installed.

Reply:
Plan! I'll `mock` the calls and build those extra tests too, on Monday though :beer: